src/dividat-driver/server/main.go

```diff
@@ -107,3 +107,25 @@ func Start(logger *logrus.Logger) context.CancelFunc {
 
 	return cancel
 }
+
+// Middleware for CORS headers, to be applied to any route that should be accessible from browser apps.
+func corsHeaders(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if len(r.Header["Origin"]) == 1 && isPermissibleOrigin(r.Header["Origin"][0]) {
+			w.Header().Set("Access-Control-Allow-Origin", r.Header["Origin"][0])
+			w.Header().Set("Access-Control-Allow-Private-Network", "true")
+		}
+
+		if r.Method == "OPTIONS" {
+			w.WriteHeader(200)
+			return
+		} else {
+			next.ServeHTTP(w, r)
+		}
+	})
+}
+
+func isPermissibleOrigin(origin string) bool {
+	isMatch, err := regexp.MatchString("\\A(http://(127\\.0\\.0\\.1|localhost)(:\\d+)?|https://(.*\\.)?dividat\\.(com|ch))\\z", origin)
```

Comment:
Do you have a particular attack scenario in mind to guard against here? If an attacker has something running on localhost or can inject malicious files there, they probably don't need to go through the user's web browser to access the driver (or worse). In the other cases, we probably have a machine running exactly one of the whitelisted hosts, and it seems a lot of effort for little gain to both put malicious code on one of the hosts and change the target of PlayOS.

I did not like the built-in list, however. I went ahead and [made it a default list](https://github.com/dividat/driver/pull/115/commits/a5be70bbd527c7339a33d052ba35b9608519f787), that can be overridden so that either alternative hosts or just a single host may be used.

Reply:
Yes, this seems to be ongoing work in the standard: https://wicg.github.io/private-network-access/#integration-websockets